### ISSUE TITLE
Check if a schema exists before running "create schema if not exists"

### ISSUE THREAD

**Issue by saugerDecathlon** [databricks/dbt-databricks#464]
### Describe the bug
When adding a "catalog" parameter to a dbt profile, the command create schema if not exists `catalog`.`schema` will always be executed even if the schema already exists. If the user does not have the right to create on catalog, he will have an error. When not specifying the catalog, the "show databases" and then "show schemas" is executed which avoid running  the "create schema if not exists". 

### Steps To Reproduce
1- Create manually a Schema (e.g. schema_test) in a catalog (e.g. catalog_test)
2- Remove right to Create on Catalog for a user
3 - Create a dbt profile with 
      schema: schema_test
      catalog: catalog_test
4- dbt run
5- Check the error

### Expected behavior
If a catalog parameter is in the profile, check if the schemas exists before running the "create schema if not exists".

**Comment by benc-db** [comment]
Thank you for reporting this issue; agreed that the current behavior is undesirable.

**Comment by tom-rb** [comment]
**TL;DR** It's a catalog quoting problem, it changes the behavior of [databricks-sql-connector](https://github.com/databricks/databricks-sql-python) package when listing schemas in catalogs.

**Workaround**
Disabling quoting on the database (aka catalog) level worked for me. E.g.
```yml
# dbt_project.yml
quoting:
  database: false
```

Hi, I have the same problem, and I could trace it until `Cursor.schemas(catalog_name=None, schema_name=None)` in `databricks-sql-connector` package.

Following the package README example, I get:
```
import os
from databricks import sql

host = os.getenv("DATABRICKS_HOST")
http_path = os.getenv("DATABRICKS_HTTP_PATH")
access_token = os.getenv("DATABRICKS_TOKEN")

connection = sql.connect(server_hostname=host, http_path=http_path, access_token=access_token)
cursor = connection.cursor()

print("> schemas('dev')")
cursor.schemas('dev')  # unquoted name, I get (default, my_schema, global_temp)
result = cursor.fetchall()
for row in result:
  print(row)

print("> schemas('`dev`')")
cursor.schemas('`dev`') # quoted name, I get (global_temp) only
result = cursor.fetchall()
for row in result:
  print(row)

cursor.close()
connection.close()
```
 
I created https://github.com/databricks/databricks-sql-python/issues/245 for further investigation.

**Comment by saugerDecathlon** [comment]
> Disabling quoting on the database (aka catalog) level worked for me. E.g.

Hi @tom-rb, thank you for your help. I have tested your workaround but it does not seem to fix my issue. I still can observe the create schema if not exists `catalog`.`schema` being executed on Databricks when adding "quoting: database: false". Am I missing something ?

**Comment by tom-rb** [comment]
Hi @saugerDecathlon, trying to isolate the issue, have you executed the example I posted in the linked 245 issue in databricks-sql-python project? You can adapt that snippet to test if your configured user or service account can actually list the schemas under a catalog.

If you're getting the list of schemas back, it may be something with your dbt configuration that's not taking the unquoting config into account. If you're not getting a list of schemas, it's more likely to be a permission issue. In any case, it's probably useful to add the dbt version you're running in the original post here (I'm assuming is a fairly recent one).
